### PR TITLE
[Fixes #14] Workflow for Pytest

### LINKED ISSUE #14: Github Actions for pytest
## Feature Request

Is your feature request related to an existing [Issue](https://www.github.com/Auto-DL/Generator/issues) or a new implementation?

No.

### Feature Description

Add github actions workflow to run the tests and generate a coverage report.

### PR DESCRIPTION
**Pull request by bamblebam**
## Pull Request

fixes #14 

### Description

Adds github workflows to check if all the pytest tests are working properly.
Had to comment out the pywin32 dependency in requirements.txt because apparently it cannot be installed properly.

### Checklist:
- [x] Followed the guidelines mentioned in our [Contributing document](../CONTRIBUTING.md)
- [x] Ensured that there aren't other open [Pull Requests](https://www.github.com/Auto-DL/Generator/pulls) for the same update/change
- [x] This PR is linked to an [Issue](https://www.github.com/Auto-DL/Generator/issues) and the issue number has been mentioned in the title and body of the PR
- [x] The code is clean and docstrings have been added or updated as required
- [x] Code is linted/formatted locally prior to submission (using `black`)

### CODE REVIEW

**Review by ADI10HERO — CHANGES_REQUESTED** [review]
Same as the main repo, please add caching and remove the "unwanted" dependencies from req.txt

File: tests/data_scraping/test_tensorflow_data.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def test_get_data_from_repository(get_test_repository_url, get_driver, tmpdir):
     tf_data.get_data_from_repository(
         get_test_repository_url, get_driver, startTimeForUrl, tmpdir
     )
-    assert len(os.listdir(tmpdir)) == 3
+    assert len(os.listdir(tmpdir)) == 27
```
Comment (ADI10HERO) [inline tests/data_scraping/test_tensorflow_data.py:52]: Your GitHub repo's size is going to change again?

File: requirements.txt [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # requirements-dev
 pre-commit==2.13.0
-black
+# black
```
Comment (ADI10HERO) [inline requirements.txt:3]: We need black right?

File: .github/workflows/testing.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+name: "Pytest"
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
```
Comment (ADI10HERO) [inline .github/workflows/testing.yml:8]: Can you modify this similar to the Main repo?
(In future if we have many branches and PRs on them)

**Review by ADI10HERO — CHANGES_REQUESTED** [review]

File: tests/data_scraping/test_tensorflow_data.py (hunk 1 above)
Comment (bamblebam) [reply tests/data_scraping/test_tensorflow_data.py:52]: Just did this to take into account the multiprocessing thing. Should'nt change again infuture or we can create a new dummy repo.

File: .github/workflows/testing.yml (hunk 3 above)
Comment (bamblebam) [reply .github/workflows/testing.yml:8]: ok I will do it

File: requirements.txt (hunk 2 above)
Comment (bamblebam) [reply requirements.txt:3]: it is listed twice so commented once because it was giving issues

File: .github/workflows/testing.yml (hunk 3 above)
Comment (bamblebam) [reply .github/workflows/testing.yml]: done

File: requirements.txt (hunk 2 above)
Comment (ADI10HERO) [reply requirements.txt:3]: Oohh then lets just remove once

File: tests/data_scraping/test_tensorflow_data.py (hunk 1 above)
Comment (ADI10HERO) [reply tests/data_scraping/test_tensorflow_data.py:52]: Cool cool, if it won't change then cool else a dummy repo would be ideal

File: requirements.txt (hunk 2 above)
Comment (bamblebam) [reply requirements.txt]: done

**Review by ADI10HERO — APPROVED** [review]

**Pull request merged**